pkg/embeddedcluster/upgrade.go

```diff
@@ -0,0 +1,381 @@
+package embeddedcluster
+
+import (
+	"bufio"
+	"context"
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/mholt/archiver/v3"
+	embeddedclusterv1beta1 "github.com/replicatedhq/embedded-cluster-kinds/apis/v1beta1"
+	embeddedclustertypes "github.com/replicatedhq/embedded-cluster-kinds/types"
+	dockerregistrytypes "github.com/replicatedhq/kots/pkg/docker/registry/types"
+	"github.com/replicatedhq/kots/pkg/imageutil"
+	"github.com/replicatedhq/kots/pkg/k8sutil"
+	registrytypes "github.com/replicatedhq/kots/pkg/registry/types"
+	"github.com/replicatedhq/kots/pkg/util"
+	kotsv1beta1 "github.com/replicatedhq/kotskinds/apis/kots/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"oras.land/oras-go/v2/registry/remote/auth"
+	"oras.land/oras-go/v2/registry/remote/credentials"
+	k8syaml "sigs.k8s.io/yaml"
+)
+
+const (
+	clusterUpgradeTimeout = 15 * time.Minute
```

Comment:
could we base this on the number of nodes somehow?

Reply:
hmm, so this timeout is for distributing the infra artifacts and upgrading ECO. @emosbaugh thoughts?

Reply:
I think we should remove this and find a logical way to cancel the operation through this context

https://github.com/replicatedhq/kots/pull/4756/files/e7559fa89645292447dad8fda47473da02e44543#diff-96da550adbaf3fbec7af8856e898e960a8ec9d2be4600202dc1a19f46f7d7d3fR132

Reply:
removed